src/brevitas_examples/llm/llm_quant/awq/pre_quant.py

```diff
@@ -96,81 +161,74 @@ def __init__(self, module):
 
         def forward(self, inp, **kwargs):
             inps.append(inp)
-            layer_kwargs.update(kwargs)
+            block_kwargs.update(kwargs)
             raise ValueError  # early exit to break later inference
 
     # patch layer 0 to catch input and kwargs
-    layers[0] = Catcher(layers[0])
+    blocks[0] = Catcher(blocks[0])
```

Comment:
I don't think we need this part of the codebase, why can't we do what we do in GPTQ to catch the input to the first block?
We can also move that piece of code to some utils in exmples/common/generative